xm-commons-logging/src/main/java/com/icthh/xm/commons/logging/util/LogObjectPrinter.java

```diff
@@ -301,19 +301,28 @@ public static String printCollectionAware(final Object object, final boolean pri
         }
 
         if (object == null) {
-            return String.valueOf(object);
+            return "null";
         }
 
         Class<?> clazz = object.getClass();
         if (!Collection.class.isAssignableFrom(clazz)) {
-            return String.valueOf(object);
+            return printTypeAware(object);
         }
 
-        return new StringBuilder().append("[<")
-                                  .append(clazz.getSimpleName())
-                                  .append("> size = ")
-                                  .append(Collection.class.cast(object).size()).append("]")
-                                  .toString();
+        return "[<"
+               + clazz.getSimpleName()
+               + "> size = "
+               + ((Collection) object).size() + "]";
+    }
+
+    private static String printTypeAware(Object value) {
+        if (value == null) {
+            return null;
+        }
+        if (value.getClass().isArray()) {
+            return Arrays.toString((Object[]) value);
```

Comment:
how about long arrays, meybe better to cut it to maybe 10 element maximum?

Reply:
I suggest to print only size.

Reply:
@sergeysenja1992 @major13ua please check